helix-view/src/document.rs

```diff
@@ -130,24 +130,41 @@ impl Document {
     // TODO: passing scopes here is awkward
     // TODO: async fn?
     pub fn load(path: PathBuf, scopes: &[String]) -> Result<Self, Error> {
-        use std::{env, fs::File, io::BufReader};
+        use std::{
+            env,
+            fs::File,
+            io::{BufReader, Cursor},
+        };
         let _current_dir = env::current_dir()?;
 
-        let file = File::open(path.clone()).context(format!("unable to open {:?}", path))?;
-        let doc = Rope::from_reader(BufReader::new(file))?;
-
-        // TODO: create if not found
+        let doc = if !path.exists() {
+            Rope::from_reader(Cursor::new(vec![]))
+        } else {
+            Rope::from_reader(BufReader::new(
+                File::open(&path).context(format!("unable to open {:?}", path))?,
+            ))
+        }?;
 
         let mut doc = Self::new(doc);
 
+        let path = if !path.exists() {
```

Comment:
```rust
let path = canonicalize_path(path)?;

fn canonicalize_path(path: &Path) -> std::io::Result<PathBuf> {
    let current_dir = std::env::current_dir()?;
    normalize_path(&current_dir.join(path))
}
```

No `if` necessary, replace the code in set_path with this new function too https://github.com/helix-editor/helix/blob/7761c88d6130364915daa23115d2ee1234b2bab5/helix-view/src/document.rs#L222-L224